block/src/block.rs

```diff
@@ -35,6 +35,20 @@ pub struct Block {
     pub ommers: Vec<Header>,
 }
 
+impl Block {
+    pub fn is_basic_valid(&self) -> bool {
+        if transactions_root(&self.transactions) != self.header.transactions_root {
```

Comment:
Is it ever possible that `self.header` is nil?

Reply:
Nope. Guaranteed by Rust's type system~~

Reply:
To get nil one must use the Option type explicitly. And if ever happen that one forgot to handle nil, the compiler would refuse the comply. :)

Reply:
That's pretty awesome.